src/Elastic.Markdown/Myst/CodeBlocks/CodeBlockArguments.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Elastic.Markdown.Diagnostics;
+
+namespace Elastic.Markdown.Myst.CodeBlocks;
+
+
+internal enum CodeBlockArgument
+{
+	Callouts,
+	Subs
+}
+
+public class InvalidCodeBlockArgumentException(string message) : ArgumentException(message);
+
+public class CodeBlockArguments
+{
+	public static bool TryParse(string args, [NotNullWhen(true)] out CodeBlockArguments? codeBlockArgs)
+	{
+		codeBlockArgs = null;
+
+		Dictionary<CodeBlockArgument, bool> arguments = [];
+
+		if (string.IsNullOrWhiteSpace(args))
+		{
+			codeBlockArgs = new CodeBlockArguments(ImmutableDictionary<CodeBlockArgument, bool>.Empty);
+			return true;
+		}
+
+		foreach (var i in args.Split(","))
```

Comment:
In .NET 9 we can now do allocation free splits on spans using `Range`'s. 

https://medium.com/codenx/net-9-readonlyspan-char-split-947d6e1211e6

Just an FYI not critical to update the PR for :)

Reply:
https://github.com/elastic/docs-builder/pull/699/files/7ebd1f7a19b9009f436b9f855ddfeda63e0e0b2b..d3e464b9a98defb8c4fa9d57315b5a81c1330668